api/src/wfl/service/firecloud.clj

```diff
@@ -126,6 +127,81 @@
                               {:Content/type "text/tab-separated-values"
                                :entities     (slurp file)})})))
 
+(defn import-entity-set
+  "
+  Formats existing sample ENTITIES into a Terra-compatible tsv
+  and uploads to WORKSPACE as an entity set entry identified by SET-NAME.
+
+  All entity references should correspond to existing entities.
+  All entity references should be of the same type.
+
+  Parameters
+  ----------
+    workspace - Fully-qualified Terra Workspace in which the entity set
+                will be created
+    set-name  - Unique identifier for the entity set
+    entities  - List of entity [Type Name] pairs
+
+  Example
+  -------
+    (import-entity-set \"workspace-namespace/workspace-name\"
+                       \"c25ee04b-...\"
+                       [[\"sample\" \"NA12878\"] [\"sample\" \"NA12879\"]])
+  "
+  [workspace set-name [[entity-type _] & _ :as entities]]
+  ;; TODO: Should we enforce common entity-type for all entities?
+  (let [columns [entity-type entity-type]
+        rows (for [[_ entity-name] entities] [set-name entity-name])]
+    (->> (util/columns-rows->terra-tsv :membership columns rows)
+         .getBytes
+         (import-entities workspace))))
+
+(defn consolidate-entities-to-set
+  "
+  Optionally consolidates ENTITIES into an entity set.
+  Returns the [Type Name] pair for the resulting entity,
+  or the passed-in entity if it was the only one specified.
+
+  Parameters
+  ----------
+    workspace - Fully-qualified Terra Workspace in which the entity set
+                would be created
+    entities  - List of entity [Type Name] pairs
+
+  Example
+  -------
+    (consolidate-entities-to-set \"workspace-namespace/workspace-name\"
+                                 [[\"sample\" \"NA12878\"] [\"sample\" \"NA12879\"]])
+  "
+  [workspace [entity & maybe-entities :as entities]]
+  ;; TODO: Should we enforce common entity-type for all entities?
+  (if (seq maybe-entities)
+    (let [id (str (UUID/randomUUID)) ; TODO: do we have access to a workload-affiliated UUID?
+          response (import-entity-set workspace id entities)]
+      [(:body response) id])
+    entity))
+
+(defn create-submission-for-entity-set
+  "
+  Optionally consolidates ENTITIES into an entity set and uses it to create a
+  submission in WORKSPACE with METHODCONFIG.
+
+  At least one entity should be specified.
+
+  Parameters
+  ----------
+    workspace    - Fully-qualified Terra Workspace in which the submission
+                   will be created.
+    methodconfig - Fully-qualified method configuration.
+    entities     - [Optional] Entity [Type Name] pairs
+  "
+  [workspace methodconfig & entities]
+  ;; TODO: refactor create-submission to support entity set specification.
+  ;; TODO: instead of variadic approach, should we take in a list of entities?
```

Comment:
Flagging this: I wasn't sure how this method will be called.  The variadic declaration allows for backwards compatibility when integrating with existing `create-submission`.  But I also thought it was possible this would instead be easier to call with a list of entities.

Reply:
Hmm - I think we only need one endpoint wrapper for submissions since there is only one. That said, this code is definitely useful and will probably live in the `covid` namespace @rexwangcc is working on. I'd put it in the test as a demonstrator for HOW you might bundle a collection of entities into one for submission.

Reply:
As a separate comment, I'd also add the name of the entity you want to create as an input because we'd probably want to append the set to an existing table in the workspace.

Reply:
And yes, I'd take a list of entities!

Reply:
All noted.  Yes, my ultimate intention in this PR is to consolidate to a single endpoint, and now that I know how it will be called that helps!

Reply:
- Method signature updated to accept a list of entities
- Integration test added to model how to call this method
- RE: " I'd also add the name of the entity you want to create as an input because we'd probably want to append the set to an existing table in the workspace" -- I believe I've addressed this by allowing the entity set name to be optionally specified, this proved useful for writing tests too.  LMK if I've misunderstood.